synapse/app/_base.py

```diff
@@ -499,6 +500,7 @@ def reload_cache_config(config: HomeServerConfig) -> None:
             previous_cache_config.__dict__,
             config.caches.__dict__,
         )
+        synapse.util.caches.TRACK_MEMORY_USAGE = config.caches.track_memory_usage
```

Comment:
Hmm, so this is safe to update to `False`, but I'm not 100% sure its safe to update this to `True`. At the very least the metrics will be wrong for a while, as we won't track memory usage for anything that is already in the cache.

Happy for us to do one of:

1. Not update this
2. Only update this to `False`
3. Check its safe to update this to `True` and add a big comment about the fact it will underestimate the cache size for a while

Reply:
I'll go for option 1 and update the docstring.

I'll also update the docs to only say that SIGHUP reloads cache factors.

Reply:
Fair!

Reply:
One more look over please?